envoy/cilium_network_policy.h

```diff
@@ -0,0 +1,218 @@
+	for (const auto& header: rule.headers()) {
+	  headers_.emplace_back(header);
+	  const auto& header_data = headers_.back();
+	  ENVOY_LOG(trace, "Cilium L7 HttpNetworkPolicyRule(): HeaderData {}={} ({})",
+		    header_data.name_.get(), header_data.value_, header_data.is_regex_ ? "regex" : "literal");
+	}
+      }
+
+      bool Matches(const Envoy::Http::HeaderMap& headers) const {
+	// Empty set matches any headers.
+	return Envoy::Router::ConfigUtility::matchHeaders(headers, headers_);
+      }
+
+      std::vector<Envoy::Router::ConfigUtility::HeaderData> headers_; // Allowed if empty.
+    };
+    
+    class PortNetworkPolicyRule : public Logger::Loggable<Logger::Id::config> {
+    public:
+      PortNetworkPolicyRule(const cilium::PortNetworkPolicyRule& rules) {
+	for (const auto& remote: rules.remote_policies()) {
+	  ENVOY_LOG(trace, "Cilium L7 PortNetworkPolicyRule(): Allowing remote {}", remote);
+	  allowed_remotes_.emplace(remote);
+	}
+	for (const auto& http_rule: rules.http_rules().http_rules()) {
+	  http_rules_.emplace_back(http_rule);
+	}
+      }
+
+      bool Matches(uint64_t remote_id, const Envoy::Http::HeaderMap& headers) const {
+	// Remote ID must match if we have any.
+	if (allowed_remotes_.size()) {
+	  bool matches = false;
+	  for (const auto& remote: allowed_remotes_) {
+	    if (remote == remote_id) {
+	      matches = true;
+	      break;
+	    }
+	  }
+	  if (!matches) {
+	    return false;
+	  }
+	}
+	// Empty set matches any payload
+	if (http_rules_.size() == 0) {
+	  return true;
+	}	
+	for (const auto& rule: http_rules_) {
+	  if (rule.Matches(headers)) {
+	    return true;
+	  }
+	}
+	return false;
+      }
+
+      std::unordered_set<uint64_t> allowed_remotes_; // Everyone allowed if empty.
+      std::vector<HttpNetworkPolicyRule> http_rules_; // Allowed if empty, but remote is checked first.
+    };
+
+    class PortNetworkPolicyRules : public Logger::Loggable<Logger::Id::config> {
+    public:
+      PortNetworkPolicyRules(const google::protobuf::RepeatedPtrField<cilium::PortNetworkPolicyRule>& rules) {
+	if (rules.size() == 0) {
+	    ENVOY_LOG(trace, "Cilium L7 PortNetworkPolicyRules(): No rules, will allow everything.");
+	}
+	for (const auto& it: rules) {
+	  rules_.emplace_back(PortNetworkPolicyRule(it));
+	}
+      }
+
+      bool Matches(uint64_t remote_id, const Envoy::Http::HeaderMap& headers) const {
+	// Empty set matches any payload from anyone
+	if (rules_.size() == 0) {
+	  return true;
+	}
+	for (const auto& rule: rules_) {
+	  if (rule.Matches(remote_id, headers)) {
+	    return true;
+	  }
+	}
+	return false;
+      }
+
+      std::vector<PortNetworkPolicyRule> rules_; // Allowed if empty.
+    };
+    
+    class PortNetworkPolicy : public Logger::Loggable<Logger::Id::config> {
+    public:
+      PortNetworkPolicy(const google::protobuf::RepeatedPtrField<cilium::PortNetworkPolicy>& rules) {
+	for (const auto& it: rules) {
+	  // TODO: Handle Missing Protocol
```

Comment:
I think there's nothing to do.
I would just comment that we ignore UDP rules because they are not relevant to an HTTP filter.

Reply:
Done.